server/services/mi-light/lib/light/light.setValue.js

```diff
@@ -0,0 +1,60 @@
+const { DEVICE_FEATURE_TYPES } = require('../../../../utils/constants');
+const logger = require('../../../../utils/logger');
+const { parseExternalId } = require('../utils/parseExternalId');
+const { NotFoundError } = require('../../../../utils/coreErrors');
+
+/**
+ * @description Change value of a milight zone
+ * @param {Object} device - The device to control.
+ * @param {Object} deviceFeature - The binary deviceFeature to control.
+ * @param {string|number} value - The new value.
+ * @example
+ * setValue(device, deviceFeature, value);
+ */
+async function setValue(device, deviceFeature, value) {
+  const { zoneId, bridgeMac } = parseExternalId(device.external_id);
+  // transform mac : adding ":"
+  let macAddressStdFormat = '';
+  for (let i = 0; i < 11; i += 2) {
+    macAddressStdFormat += bridgeMac.slice(i, i + 2);
+    if (i !== 10) {
+      macAddressStdFormat += ':';
+    }
+  }
+  const bridge = this.bridgesByMac.get(macAddressStdFormat.trim());
+
+  if (!bridge) {
+    throw new NotFoundError(`BRIDGE_NOT_FOUND`);
+  }
+  logger.info(
+    `Connecting to milight bridge : Mac = "${bridgeMac}", ip = ${bridge.ip}, type = ${bridge.type}, name = ${bridge.name}`,
+  );
+  logger.debug(
+    `Changing state of light ${device.external_id} with IP ${bridge.ip} . Zone = ${zoneId}. New value = ${value}`,
+  );
+
+  // const commands = await this.milightClient.commandsV6;
+  // const MiLight = await this.milightClient.MilightController;
+  const commands = require('node-milight-promise').commandsV6;
+  const MiLight = require('node-milight-promise').MilightController;
```

Comment:
Can't it be required at the top of the file ?

Reply:
Yes I think.

Reply:
Those require should be in the `index.js` file of the service and injected in the handler class.

Example: 

https://github.com/GladysAssistant/Gladys/blob/master/server/services/philips-hue/index.js#L8